src/main/java/io/kestra/plugin/redis/ListPush.java

```diff
@@ -65,36 +67,30 @@ public class ListPush extends AbstractRedisConnection implements RunnableTask<Li
     public Output run(RunContext runContext) throws Exception {
         try (RedisFactory factory = this.redisFactory(runContext)) {
 
-            Integer count = 0;
-            if (this.from instanceof String || this.from instanceof List) {
-                Flowable<Object> flowable;
-                Flowable<Integer> resultFlowable;
-                if (this.from instanceof String) {
-                    URI from = new URI(runContext.render((String) this.from));
-                    try (BufferedReader inputStream = new BufferedReader(new InputStreamReader(runContext.uriToInputStream(from)))) {
-                        flowable = Flowable.create(FileSerde.reader(inputStream), BackpressureStrategy.BUFFER);
-                        resultFlowable = this.buildFlowable(flowable, runContext, factory);
-
-                        count = resultFlowable
-                            .reduce(Integer::sum)
-                            .blockingGet();
-                    }
-                } else {
-                    flowable = Flowable.fromArray(((List<Object>) this.from).toArray());
-                    resultFlowable = this.buildFlowable(flowable, runContext, factory);
-
+            Integer count;
+            if (this.from instanceof String fromUrl) {
+                URI from = new URI(runContext.render(fromUrl));
+                try (BufferedReader inputStream = new BufferedReader(new InputStreamReader(runContext.uriToInputStream(from)))) {
+                    Flowable<Object> flowable = Flowable.create(FileSerde.reader(inputStream), BackpressureStrategy.BUFFER);
+                    Flowable<Integer> resultFlowable = this.buildFlowable(flowable, runContext, factory);
                     count = resultFlowable
                         .reduce(Integer::sum)
                         .blockingGet();
-                    runContext.metric(Counter.of("records", count));
                 }
+            } else if (this.from instanceof List<?> fromList) {
+                Flowable<Object> flowable = Flowable.fromArray((fromList).toArray());
+                Flowable<Integer> resultFlowable = this.buildFlowable(flowable, runContext, factory);
+                count = resultFlowable
+                    .reduce(Integer::sum)
+                    .blockingGet();
+            }
+            else {
+                // should not occur as validation mandates String or List
+                throw new IllegalVariableEvaluationException("Invalid 'from' property type :" + from.getClass());
             }
 
-            Output output = Output.builder().count(count).build();
-
-            factory.close();
```

Comment:
The factory is an autoclosable, so it will be closed automatically at the end of the try-with-ressource:

```java
try (RedisFactory factory = this.redisFactory(runContext)) {
}
```

Reply:
So the manuall call to close was not needed (and in fact masy lead to issues)